src/server/detail/save_stages_controller.cc

```diff
@@ -338,9 +338,11 @@ void SaveStagesController::CloseCb(unsigned index) {
   if (auto& snapshot = snapshots_[index].first; snapshot) {
     shared_err_ = snapshot->Close();
 
-    lock_guard lk{rdb_name_map_mu_};
+    unique_lock lk{rdb_name_map_mu_};
     for (const auto& k_v : snapshot->freq_map())
       rdb_name_map_[RdbTypeName(k_v.first)] += k_v.second;
+    lk.unlock();
+    snapshot.reset();
```

Comment:
can you explain wh we need to call reset here? why reset on destructor leads to crush?

Reply:
because our memory allocator is thread-local and RdbSnapshot uses transitively the shard allocator. 
I succeeded to see the stacktrace with a differrent configuration that lead me to the root cause.